### ISSUE TITLE
Error `RangeError: Maximum call stack size exceeded`

### ISSUE THREAD

**Issue by amantiwari1** [TanStack/table#4090]
### Describe the bug

After working migrated v7 to v8 

I tested tsc and suddenly error comes

```
$ D:\work\****\****\node_modules\.bin\tsc
D:\work\****\****\node_modules\typescript\lib\tsc.js:96888
                throw e;
                ^

RangeError: Maximum call stack size exceeded
    at isRelatedTo (D:\work\****\****\node_modules\typescript\lib\tsc.js:53941:33)    at someTypeRelatedToType (D:\work\****\****\node_modules\typescript\lib\tsc.js:54272:35)
    at unionOrIntersectionRelatedTo (D:\work\****\****\node_modules\typescript\lib\tsc.js:54208:24)
    at structuredTypeRelatedTo (D:\work\****\****\node_modules\typescript\lib\tsc.js:54509:34)
    at recursiveTypeRelatedTo (D:\work\****\****\node_modules\typescript\lib\tsc.js:54431:30)
    at isRelatedTo (D:\work\****\****\node_modules\typescript\lib\tsc.js:54019:25)    at typeRelatedToEachType (D:\work\****\****\node_modules\typescript\lib\tsc.js:54257:35)
    at unionOrIntersectionRelatedTo (D:\work\****\****\node_modules\typescript\lib\tsc.js:54197:28)
    at structuredTypeRelatedTo (D:\work\****\****\node_modules\typescript\lib\tsc.js:54509:34)
    at recursiveTypeRelatedTo (D:\work\****\****\node_modules\typescript\lib\tsc.js:54431:30)
error Command failed with exit code 1.
```

I confirmed it is related to react table 
because I commented on it and it works 

But unable to figure to fix it

here's code 

```js
  const tableReact = useReactTable({
    data,
    columns,
    getCoreRowModel: getCoreRowModel(),
    getSortedRowModel: getSortedRowModel(),
    state: {
      sorting,
    },
    onSortingChange: setSorting,
  })
```

### Your minimal, reproducible example

no terminal support :/ on this platform

### Steps to reproduce

1. migrate v7 to v8
2. all table is working well
3. type check - `npx tsc` or `yarn tsc` 
4. RangeError: Maximum call stack size exceeded

### Expected behavior

no error :)

### How often does this bug happen?

Only once

### Platform

OS - Windows 11
Edge - 103.0.1264.44
 

### react-table version

8.1.3

### TypeScript version

4.4

### Terms & Code of Conduct

- [X] I agree to follow this project's Code of Conduct
- [X] I understand that if my bug cannot be reliable reproduced in a debuggable environment, it will probably not be fixed and this issue may even be closed.

**Comment by KevinVandy** [comment]
A code sandbox reproducing this is probably the only way this could be debugged. Though look into making sure your data is memoized or stable

**Comment by amantiwari1** [comment]
I understand! finally, I made it - https://github.com/amantiwari1/react-table-bugs


code sandbox - https://codesandbox.io/s/headless-violet-yskpzy

there is no terminal in the code sandbox  so 

for debugging, you might clone https://github.com/amantiwari1/react-table-bugs

1) yarn 
2) yarn tsc 

error come

**Comment by amantiwari1** [comment]
@KevinVandy  any update?

**Comment by PerryRylance** [comment]
@amantiwari1  I'm encountering the same issue today and working through solving it. Have you tried starting with the example at https://react-table-v7.tanstack.com/docs/quick-start and building on that?

**Comment by amantiwari1** [comment]
@PerryRylance  version 7 works fine 
I am using react table v8 actually

**Comment by amantiwari1** [comment]
@tannerlinsley do you have any idea how to fix this bug?

**Comment by tannerlinsley** [comment]
The codesandbox supplied here is not a reproducible example. It appears to just be a dump of a repo into code sandbox. In order for me to help debug this, I need something I can open and run right away and see an error. 

Also, does the code work at runtime, but not during typechecking? I don't understand why you are seeing this error when you run `tsc`.

If it's not working at runtime, my best guess here is that you are not memoizing something like `data` or `columns`. Outside of that I'll need a better codesandbox to debug. Do not just dump your code into a sandobox repo. It's much better if you take one of the existing codesandbox examples from the repo and make only the required changes to make it break similar to your own.

Please reopen a new issue with the codesandbox repro and I'll take a deeper look. Thanks!

**Comment by RafaelGB** [comment]
I can confirm that with 8.2.1 the problem was fixed. Thanks!
